src/Cassandra/DataStax/Graph/TinkerpopDate.cs

```diff
@@ -0,0 +1,150 @@
+﻿//
+//      Copyright (C) DataStax Inc.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+using System;
+
+namespace Cassandra.DataStax.Graph
+{
+    /// <summary>
+    /// Represents a value of type "Date" in Tinkerpop. It's basically a wrapper around a <see cref="DateTimeOffset"/> value.
+    /// </summary>
+    public struct TinkerpopDate : IEquatable<TinkerpopDate>, IEquatable<DateTimeOffset>, IEquatable<DateTime>
```

Comment:
The issue is with serialization, I actually don't know if TK's Date is even being mapped to any CQL type so this was more of a thing to be on the safe side in case a user needs to provide an object of type "Date" since `DateTimeOffset` will always be serialized as `Timestamp`.

Reply:
We can consider TinkerPop to be a mapping layer, so it use a type according to its mapping rules.

From our side, having those 2 types mapped to just one is fine. The type system in TK is huge but we don't necessarily have to replicate that. I think it will be more comfortable for users to use `DateTimeOffset` for both.

Reply:
But if the DB is expecting a `Date` for a particular property then the user will never be able to provide a value for it using the driver because he driver will always serialize it as `Timestamp`.

Reply:
Since I actually didn't find any scenario where the `Date` type was expected by the DB I think I'll just remove this type, remove the serializer and keep the deserializer but change it to return a `DateTimeOffset`.

Reply:
Irt `TinkerpopTimestamp` I'm not so sure that the DB doesn't expect these, I'll have to run some experiments.

Reply:
One option that we have is to make the deserializers return `DateTimeOffset` instances but keep the serializers and document users that if they get an error saying that the DB expects `Date` or `Timestamp` they can use the utility methods that the driver provides to wrap a `DateTimeOffset` value into the right TK type.

Reply:
I think we can have 1 serializer, that deserializes both `g:Date` and `g:Timestamp` into a `DateTimeOffset`, and always serializes `g:Timestamp` (treating TK's `Date` as synonym of `Timestamp`).

Reply:
there's also `gx:Instant` which is what the DB expects in GraphSON3 (I need to double check what type it expects in GraphSON2) so using that approach it would be one serializer that deserializes all 3 TK types to `DateTimeOffset` and always serializes `DateTimeOffset` to `gx:Instant`. I know that they are all supposed to be synonyms but I do remember that I was getting failures if I didn't provide the correct type that the DB expects (which I eventually found out that it was `gx:Instant`).

Reply:
I removed `TinkerpopTimestamp` and `TinkerpopDate`, now all 3 types deserialize to `DateTimeOffset` and `DateTimeOffset` is serialized to `gx:Instant`

Reply:
Sounds good